gems/aws-sdk-core/lib/aws-sdk-core/assume_role_credentials.rb

```diff
@@ -3,25 +3,28 @@
 require 'set'
 
 module Aws
-
-  # An auto-refreshing credential provider that works by assuming
-  # a role via {Aws::STS::Client#assume_role}.
+  # A credential provider that assumes a role via
+  # {Aws::STS::Client#assume_role}.
   #
   #     role_credentials = Aws::AssumeRoleCredentials.new(
   #       client: Aws::STS::Client.new(...),
   #       role_arn: "linked::account::arn",
   #       role_session_name: "session-name"
   #     )
-  #
   #     ec2 = Aws::EC2::Client.new(credentials: role_credentials)
   #
-  # If you omit `:client` option, a new {STS::Client} object will be
-  # constructed.
+  # If you omit `:client` option, a new {Aws::STS::Client} object will be
+  # constructed with additional options that were provided.
+  #
+  # Automatically handles refreshing credentials if an Expiration time is
```

Comment:
What does if an expiration time is provided in the payload mean?  This to me sounds like something a user needs to do to get this behavior.

Reply:
If AssumeRole returns Expiration in the JSON body, it's used here.

Reply:
How do I as a customer know if my credentials will be refreshing or if the expiration will be in the json body of the response?  
I'd lean towards leaving the "Auto refreshing" in the description.  Or for credentials like process credentials where the user's custom process has control over the behavior, then document it like this.